Libplanet.Tests/Common/Action/DumbAction.cs

```diff
@@ -62,16 +87,113 @@ public IAccountStateDelta Execute(IActionContext context)
                 ? $"{Item}:{context.Rehearsal}"
                 : Item;
             items = items is null ? item : $"{items},{item}";
+
+            if (RecordRandom)
+            {
+                states = states.SetState(
+                    RandomRecordsAddress,
+                    context.Random.Next()
+                );
+            }
+
             return states.SetState(TargetAddress, items);
         }
 
+        public void Render(
+            IActionContext context,
+            IAccountStateDelta nextStates)
+        {
+            if (RenderRecords.Value is null)
+            {
+                RenderRecords.Value = ImmutableList<RenderRecord>.Empty;
+            }
+
+            RenderRecords.Value = RenderRecords.Value.Add(new RenderRecord()
+            {
+                Render = true,
+                Action = this,
+                Context = context,
+                NextStates = nextStates,
+            });
+        }
+
+        public void Unrender(
+            IActionContext context,
+            IAccountStateDelta nextStates)
+        {
+            if (RenderRecords.Value is null)
+            {
+                RenderRecords.Value = ImmutableList<RenderRecord>.Empty;
+            }
+
+            RenderRecords.Value = RenderRecords.Value.Add(new RenderRecord()
+            {
+                Unrender = true,
+                Action = this,
+                Context = context,
+                NextStates = nextStates,
+            });
+        }
+
         public void LoadPlainValue(
             IImmutableDictionary<string, object> plainValue
         )
         {
             Item = (string)plainValue["item"];
             TargetAddress = new Address((byte[])plainValue["target_address"]);
             RecordRehearsal = (bool)plainValue["record_rehearsal"];
+            RecordRandom =
+                plainValue.ContainsKey("record_random") &&
+                plainValue["record_random"] is bool r &&
+                r;
+        }
+
+        public bool Equals(DumbAction other)
+        {
+            return !ReferenceEquals(null, other) && (
```

Comment:
Is this different from `!(other is null)`?

Reply:
No, it means the same.  It's generated by Rider.  I will changed it to use pattern matching syntax.

Reply:
Fixed: 10f136b87996339da9678896db76bdfa6ead6807.